### PR TITLE
Support pinning the chef vault version

### PR DESCRIPTION
**Pull request by dhui**
Pin to the latest non 3.x release https://github.com/chef/chef-vault/blob/master/Changelog.md

Chef-vault release version 3.0.0 today which broke for us.
Namely, chef-vault now requires a version of Ruby >= 2.2.0 but we're running Ruby 2.1.x

### CODE REVIEW

**Comment by brunorubin** [comment]
+1 on that, we are having exactly the same issue.

File: recipes/sumoconf.rb [hunk 1]
```diff
@@ -32,6 +32,7 @@
 
 chef_gem 'chef-vault' do
   compile_time true if respond_to?(:compile_time)
+  version node['sumologic']['chef_vault_version']
```
Comment (majormoses) [inline recipes/sumoconf.rb]: I think `version node['sumologic']['chef_vault_version'] unless node['sumologic']['chef_vault_version'].nil?` this gives people the flexibility to pin or not pin it.

File: attributes/default.rb [hunk 2]
```diff
@@ -149,3 +149,5 @@
   # Collector Restart Command
   default['sumologic']['collectorRestartCmd'] = "#{default['sumologic']['installDir']}/collector restart"
 end
+
+default['sumologic']['chef_vault_version'] = '2.9.0'
```
Comment (majormoses) [inline attributes/default.rb]: I would default this to `nil` as not everyone needs to pin right? If you need to pin you can without affecting other users.

Comment (dhui) [inline attributes/default.rb]: As requested, I've changed this to default to `nil`.

It's good practice to pin your dependencies, but this is your package so I'll follow your rules.

Comment (majormoses) [inline attributes/default.rb]: I am not actually a maintainer just a consumer, if I had been I would not even include the gem unless you are using chef vault but that's another discussion.

**Comment by majormoses** [comment]
All those failures should be fixed once they merge this #130

File: attributes/default.rb (hunk 2 above)
Comment (majormoses) [inline attributes/default.rb]: Yes I agree pinning is generally good, that being said this cookbook (as opposed to your chef install) should support 2 and 3 for some period of time before dropping 2.x support so we can't pin across multiple versions making everyone happy so this allows the consumer to specify which behavior they want and works for them.

**Comment by dhui** [comment]
Merged in master to preserve original commits. I can rebase if the maintainers prefer a cleaner commit history.

**Comment by majormoses** [comment]
@dhui in general I do prefer rebase to merge but I want to get this merged so people can be unblocked here and the history is not too bad.

**Pull request merged**